### ISSUE TITLE
AoS Feature - Consider Rabbit Hole node redirect for unauthenticated when linked directly to source (pre-save).

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa#4446]
https://iowaweb.slack.com/archives/C0164FNN1PV/p1636558112346600

## Proposed Solution

- [x] Enable rabbit hole node (already a composer dependency. HR is using for taxonomy) and configure for AoS content type with display node set as default with per node override set. Make sure all of this is captured in the AoS split.
- [x] On pre-save, if source link and direct to source are set, change the rabbit hole setting to redirect and set the url to the source link. Establish module specific to the split to handle this.
- [x] Make sure these rabbit hole settings are hidden to webmasters and below.
- [x] Update config on sites that use AoS split if necessary (education, desmoines?)

**Comment by pyrello** [comment]
Some thoughts about this:
* It would be nice to figure out how to incorporate the source link into Rabbit Hole settings per node to allow that to be used as a redirect.
* We should probably select the return code (e.g. 301, 404) for the user and hide the option if that shows on the form

**Comment by joewhitsitt** [comment]
Updated IS with proposed solution

**Comment by joewhitsitt** [comment]
I think this looks good. Supposedly, there is `hook_rabbit_hole_values_alter()` in the module that could be used to set these on the fly, but I couldn't get it to work and it was firing more often than I wanted it to...

So, webmasters can't see the rabbit hole settings based on permissions so I don't think all of the form #states logic is necessary for this. I pushed up a change that just disables it and provides a message about what is actually being done with the functionality.

I changed to a 301 instead of a temp 307 because in most cases they are redirecting to a canonical source, not a temporary detour.

I think the logic assumes these fields exist more than I like but they shouldn't be deleted any time soon. I did add an extra check for rabbbit hole functionality being available and changing the values back to the content type default if not directly linked.
